### PR TITLE
feat(inputs.amqp_consumer): Allow specification of queue arguments

### LINKED ISSUE #16140: amqp_consumer input : allow queue arguments from config
### Use Case

I need to be able to set a `max-length` attribute for the queue declared by telegraf on rabbitmq

### Expected behavior

I expect to be able to configure queue arguments.
```
queue_arguments = {"x-max-length": 100}
```

### Actual behavior

I cannot set queue arguments

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by n1b0r**
## Summary

Allow specification of queue arguments
## Checklist

   * [x]  No AI generated code was used in this PR

## Related issues

resolves #16140

### CODE REVIEW

**Comment by DStrand1** [comment]
@n1b0r Thanks for the PR! Could you sign the CLA, run `make lint` and also comment in your PR description that no AI generated code was used in this PR? Thanks!

**Comment by n1b0r** [comment]
Hello, I signed the CLA, make lint does not change anything, and I commented the PR about AI. I am not sure why some circleci tests are failing. Should I do something about it ?

File: plugins/inputs/amqp_consumer/amqp_consumer.go [hunk 1]
```diff
@@ -369,14 +370,19 @@ func (a *AMQPConsumer) declareQueue(channel *amqp.Channel) (*amqp.Queue, error)
 		queueDurable = false
 	}
 
+	queueArgs := make(amqp.Table, len(a.QueueArguments))
+		for k, v := range a.QueueArguments {
+			queueArgs[k] = v
+		}
```
Comment (DStrand1) [inline plugins/inputs/amqp_consumer/amqp_consumer.go:376]: @n1b0r I think this should fix the linter issue
```suggestion
	for k, v := range a.QueueArguments {
		queueArgs[k] = v
	}
```

**Comment by DStrand1** [comment]
Could you add this field to the sample config and readme as well? Thanks!

**Review by DStrand1 — APPROVED** [review]

**Review by srebhan — APPROVED** [review]
Thanks a lot for your contribution @n1b0r!

**Pull request merged**